deepchecks/utils/distribution/plot.py

```diff
@@ -369,3 +371,195 @@ def sort_int_and_strings(a, b):
+            fixedrange=True,
+            autorange=True,
+            rangemode='normal',
+            title='Frequency (Log Scale)',
+            type='log'
+        )
+        xaxis_layout = dict(type='category')
+
+    else:
+        x_range = (
+            min(dist.min(), dist.min()),
+            max(dist.max(), dist.max())
+        )
+
+        # Heuristically take points on x-axis to show on the plot
+        # The intuition is the graph will look "smooth" wherever we will zoom it
+        # Also takes mean and median values in order to plot it later accurately
+        xs = sorted(np.concatenate((
+            np.linspace(x_range[0], x_range[1], 50),
+            np.quantile(dist, q=np.arange(0.02, 1, 0.02))
+        )))
+
+        traces: List[go.BaseTraceType] = []
+
+        # In order to plot the common and outliers parts of the graph in different colors, we need to separate them into
+        # different traces. We do it by creating a mask for each part and then using it to filter the data.
+        # However, for the graphs to start and end smoothly, we need to add a point in the beginning and end of the
+        # common part. Those duplicate points will be set to start or end each trace in 0.
+        all_arr = [1 if lower_limit <= x <= upper_limit else 0 for x in xs]
+        common_beginning = all_arr.index(1)
+        common_ending = len(all_arr) - 1 - all_arr[::-1].index(1)
+
+        show_lower_outliers = common_beginning != 0
+        show_upper_outliers = common_ending != len(xs) - 1
+        total_len = len(xs) + show_lower_outliers + show_upper_outliers
+
+        mask_common = np.zeros(total_len, dtype=bool)
+        mask_outliers_lower = np.zeros(total_len, dtype=bool)
+        mask_outliers_upper = np.zeros(total_len, dtype=bool)
+
+        density = list(get_density(dist, xs))
+
+        # If there are lower outliers, add a duplicate point to the beginning of the common part:
+        if common_beginning != 0:
+            xs.insert(common_beginning, xs[common_beginning])
+            density.insert(common_beginning, density[common_beginning])
+            mask_outliers_lower[:common_beginning + 1] = True
+            common_ending += 1
+
+        # If there are upper outliers, add a duplicate point to the end of the common part:
+        if common_ending != len(xs) - 1:
+            xs.insert(common_ending + 1, xs[common_ending])
+            density.insert(common_ending + 1, density[common_ending])
+            mask_outliers_upper[common_ending + 1:] = True
+
+        mask_common[common_beginning + show_lower_outliers:common_ending + show_upper_outliers] = True
+
+        density_common = np.array(density) * mask_common
+        density_outliers_lower = np.array(density) * mask_outliers_lower
+        density_outliers_upper = np.array(density) * mask_outliers_upper
+
+        # Replace zeros with None so that they won't be plotted:
+        density_common = [x or None for x in density_common]
```

Comment:
What to the zeros represent?

Reply:
outliers if this is the common, and common if this is the outlier. Will make it clearer